src/main/kotlin/com/openlattice/data/ids/jobs/FixDuplicateIdAssignmentJob.kt

```diff
@@ -0,0 +1,301 @@
+                update(psDst, newId, originalId)
+                update(psEdge, newId, originalId)
+            }
+            listOf(psSyncs,psIds, psData, psEdges, psDst, psEdge).forEach { it.executeBatch() }
+
+            connection.commit()
+        }
+    }
+
+    private fun migrate(
+            ps: PreparedStatement,
+            originalId: UUID,
+            entitySetId: UUID
+    ) {
+        bind(ps, entitySetId, originalId)
+        ps.addBatch()
+    }
+
+    private fun update(
+            ps: PreparedStatement,
+            newId: UUID,
+            originalId: UUID
+    ) {
+        bind(ps, newId, originalId)
+        ps.addBatch()
+    }
+
+    private fun bind(ps: PreparedStatement, entitySetId: UUID, entityKeyId: UUID) {
+        ps.setObject(1, entitySetId)
+        ps.setObject(1, entityKeyId)
+    }
+
+    private fun updateIdAssignment(ps: PreparedStatement, entityKey: EntityKey, newId: UUID) {
+        ps.setObject(1, newId)
+        ps.setObject(2, entityKey.entitySetId)
+        ps.setString(3, entityKey.entityId)
+    }
+
+    private fun recordCollision(id: UUID, entityKeys: List<EntityKey>) {
+        hds.connection.use { connection ->
+            connection.prepareStatement(INSERT_COLLISION).use {ps ->
+                ps.setObject(1, id )
+                entityKeys.forEach {
+                    ps.setObject(2, it)
+                    ps.addBatch()
+                }
+                ps.executeBatch()
+            }
+        }
+    }
+
+    override fun initialize() {
+        state.idCount = getIdCount()
+    }
+
+    override fun updateProgress() {
+        progress = ((100 * state.idsProceessed) / state.idCount).toByte()
+    }
+
+    private fun getIdCount(): Long {
+        return hds.connection.use { connection ->
+            connection.createStatement().use { statement ->
+                val rs = statement.executeQuery(COUNT_SYNC_IDS)
+                if (rs.next()) {
+                    ResultSetAdapters.count(rs)
+                } else {
+                    throw IllegalStateException("Result set for counting ids cannot empty.")
+                }
+            }
+        }
+    }
+
+    @JsonIgnore
+    override fun setHikariDataSource(hds: HikariDataSource) {
+        this.hds = hds
+    }
+
+    @JsonIgnore
+    override fun setIdGenerationService(idService: HazelcastIdGenerationService) {
+        this.idService = idService
+    }
+}
+
+private const val DUPLICATES_FIELD = "duplicates"
+private val COUNT_SYNC_IDS = """
+SELECT count(*) FROM ${SYNC_IDS.name}
+""".trimIndent()
+
+/**
+ * Returns the duplicates for a batch of ids.
+ * 1. ids - array of uuids
+ */
+private val SELECT_NEXT_DUPLICATES = """
+SELECT  ${ID.name}, 
+        jsonb_agg(jsonb_build_object(
+            '${SerializationConstants.ENTITY_SET_ID}', ${ENTITY_SET_ID.name},
+            '${ENTITY_SET_ID.name}${SerializationConstants.ENTITY_SET_ID}${ENTITY_SET_ID.name}',${ENTITY_ID.name})
```

Comment:
what is this `${ENTITY_SET_ID.name}${SerializationConstants.ENTITY_SET_ID}${ENTITY_SET_ID.name}` value?

Reply:
following up on this one -- this looks like a funky string value

Reply:
Not sure how this came back. fixed again.